pkg/xray/resource_xray_settings.go

```diff
@@ -0,0 +1,99 @@
+package xray
+
+import (
+	"context"
+	"log"
+	"net/http"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceXraySettings() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceXrayDbSyncTimeCreate,
+		ReadContext:   resourceXrayDbSyncTimeRead,
+		UpdateContext: resourceXrayDbSyncTimeUpdate,
+		DeleteContext: resourceXrayDbSyncTimeDelete,
+		Description:   "Provides an Xray DB Sync Time resource.",
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"db_sync_updates_time": {
+				Type:             schema.TypeString,
+				Required:         true,
+				Description:      "The time of the Xray DB sync daily update job.",
+				ValidateDiagFunc: matchesHoursMinutesTime,
+			},
+		},
+	}
+}
+
+type DbSyncDailyUpdatesTime struct {
+	DbSyncTime string `json:"db_sync_updates_time"`
+}
+
+func unpackDBSyncTime(s *schema.ResourceData) DbSyncDailyUpdatesTime {
+	d := &ResourceData{s}
+	dbSyncTime := DbSyncDailyUpdatesTime{
+		DbSyncTime: d.getString("db_sync_updates_time", false),
+	}
+	return dbSyncTime
+}
+
+func packDBSyncTime(dbSyncTime DbSyncDailyUpdatesTime, d *schema.ResourceData) diag.Diagnostics {
+	if err := d.Set("db_sync_updates_time", dbSyncTime.DbSyncTime); err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func getDbSyncTime(client *resty.Client) (DbSyncDailyUpdatesTime, *resty.Response, error) {
```

Comment:
You created this function and then use it in 1 place? You've actually added 3 lines of code. Please remove this function and inline it

Reply:
removed